connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -780,6 +774,14 @@ protected void stopServices() {
         }
     }
 
+    // Timeout for herderExecutor to gracefully terminate is set to a value to accommodate
+    // reading to the end of the config topic + successfully attempting to stop all connectors and tasks and a buffer of 10s
+    private long herderExecutorTimeoutMs() {
+        return this.workerSyncTimeoutMs +
+                config.getInt(DistributedConfig.WORKER_SYNC_TIMEOUT_MS_CONFIG) +
```

Comment:
Wait a minute, is this right? Why are we using the sync timeout twice here?

Reply:
Nope that wasn't right. Changed it now. Good catch.